client/sites-dashboard/components/sites-table-row.tsx

```diff
@@ -0,0 +1,117 @@
+import styled from '@emotion/styled';
+import { useI18n } from '@wordpress/react-i18n';
+import SiteIcon from 'calypso/blocks/site-icon';
+import EllipsisMenu from 'calypso/components/ellipsis-menu';
+import PopoverMenuItem from 'calypso/components/popover-menu/item';
+
+interface SiteTableRowProps {
+	site: Site;
+}
+
+interface Site {
+	ID: number;
+	name: string;
+	slug: string;
+	URL: string;
+	plan: SitePlan;
+}
+
+interface SitePlan {
+	product_name_short: string;
+}
+
+const Row = styled.tr`
+	line-height: 2em;
+	border-bottom: 1px solid #eee;
+	td {
+		padding-top: 12px;
+		padding-bottom: 12px;
+		vertical-align: middle;
+		font-size: 14px;
+		line-height: 20px;
+		letter-spacing: -0.24px;
+		color: var( --studio-gray-60 );
+	}
+	@media only screen and ( max-width: 781px ) {
+		.sites-table-row__mobile-hidden {
+			display: none;
+		}
+	}
+`;
+
+const SiteName = styled.h2`
+	font-weight: 500;
+	font-size: 16px;
+	letter-spacing: -0.4px;
+	color: var( --studio-gray-100 );
+	margin-bottom: 8px;
+	a {
+		color: inherit;
+		&:hover {
+			text-decoration: underline;
+		}
+	}
+`;
+
+const SiteUrl = styled.a`
+	color: var( --studio-gray-60 );
+	&:visited {
+		color: var( --studio-gray-60 );
+	}
+`;
+
+const getDashboardUrl = ( slug: string ) => {
+	return '/home/' + slug;
+};
+
+const displaySiteUrl = ( siteUrl: string ) => {
+	return siteUrl.replace( 'https://', '' ).replace( 'http://', '' );
+};
+
+const VisitDashboardItem = ( site: Site ) => {
+	const { __ } = useI18n();
+	return (
+		<PopoverMenuItem href={ getDashboardUrl( site.slug ) }>
+			{ __( 'Visit Dashboard' ) }
+		</PopoverMenuItem>
+	);
+};
+
+export default function SitesTableRow( { site }: SiteTableRowProps ) {
+	const { __ } = useI18n();
+	return (
+		<Row key={ site.ID }>
```

Comment:
non-blocking: What's the use of this key? sorting purposes?

Reply:
> non-blocking: What's the use of this key? sorting purposes?

@vykes-mac Great catch! The `key=` is actually unnecessary in this context — it was copy-pasta. I've removed it in 6ef1d7ca7e

Regarding the use of `key=` generally, [here's my understanding](https://reactjs.org/docs/lists-and-keys.html#keys):

> Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside the array to give the elements a stable identity

We are using `key=` for our use of the `<SitesTableRow>` component:

https://github.com/Automattic/wp-calypso/blob/6ef1d7ca7e7df3fae7253d443e43ebb46be915ec/client/sites-dashboard/components/sites-table.tsx#L48-L52